cmake/API.cmake

```diff
@@ -31,7 +31,10 @@ set(FPRIME_AUTOCODER_TARGET_LIST "" CACHE INTERNAL "FPRIME_AUTOCODER_TARGET_LIST
 #####
 macro(restrict_platforms)
     set(__CHECKER ${ARGN})
-    if (NOT FPRIME_TOOLCHAIN_NAME IN_LIST __CHECKER AND NOT FPRIME_PLATFORM IN_LIST __CHECKER)
+    if (FPRIME_TOOLCHAIN_NAME IN_LIST __CHECKER)
```

Comment:
We might add a comment to explain all these empty if conditions.

Each of these empty if blocks are the valid-case, that is, the platform is supported.  However, the reason why this is necessary is that this function is a `macro` and not a `function`.  Macros copy-paste the code into the calling context.  Thus, all these valid cases want to avoid the `return` call on line 41.  The return call on line 41 returns from the calling context (i.e. a restricted `CMakeList.txt` will return and not process the component setup).  We do not want this return when the platform is allowed.

This pattern is great for writing a guard statement, but not so easy to understand.

Reply:
Good point. I added a comment.